### DISCUSSION TITLE
Thoughts on Quarto

### DISCUSSION THREAD

**Discussion by stefanbringuier** [sparks-baird/auto-paper#7] [Ideas]
@sgbaird  thanks for sharing your workflow for scientific writing, curious what your thoughts are on the [Quarto](https://quarto.org/) to handle many of the aspects in your diagram:

<img src="https://raw.githubusercontent.com/sparks-baird/auto-paper/main/diagram/auto-paper-new12-protocols.png" alt="drawing" width="600"/>

More generally, how do you approach using literate programming to write scientific or technical documents?

**Comment by sgbaird** [comment]
Quarto is new to me. Thanks for the suggestion! @kjappelbaum [mentioned show-your-work to me](https://github.com/sparks-baird/auto-paper/issues/6), and I'm pretty interested in trying this out. I've explored it enough to think that I want to try it out with at least one manuscript. Like @kjappelbaum mentioned, it seems better if you use it near the start of a project. I haven't picked out the specific project yet.

I've been building [this LaTeX style file](https://github.com/sparks-baird/auto-paper/blob/main/autopaper.sty) over the course of a few years. I enjoyed my experience with pandoc as it related to my [xtal2png JOSS manuscript](https://joss.theoj.org/papers/10.21105/joss.04528) (pretty sure this was pandoc). Made me realize that some of the benefits of LaTeX are there, too. Personally, I'm far too invested in LaTeX to switch to a different language (at least it would require something pretty significant to get me to fully switch). Something else I consider is a journal's set of requirements for publication. The standard is to accept LaTeX source files or a word document file, and sometimes they have more stringent rules on what's allowed for the LaTeX. Sometimes they provide templates. So far, I haven't run into too much trouble using the LaTeX style I mentioned and the `elsarticle` class.

I think Quarto could certainly be a nice tool to replace or augment a lot of the workflow mentioned here; for example, using Python scripts, pulling from BibTeX files auto-generated by Zotero, and using version control with the Quarto documents. Something @sp8rks might also be interested in.

**Comment by stefanbringuier** [comment]
Oh wow, [show-your-work](https://github.com/showyourwork/showyourwork) is definitely something I'm going to explore! Didn't know about it, thanks! The main issue for me is it looks like only `python` is supported, I need something that also supports [Julia](https://julialang.org/). I also wonder how simulation code/scripts that generate data and take considerable about of time to run are handled (i.e., if someone wanted to rerun your paper containing DFT calculations with different settings in the input script and generate the updated paper/figures once its completed). In `Julia` I use [DrWatson](https://github.com/JuliaDynamics/DrWatson.jl) to help out with this but it doesn't handle the manuscript aspect like [show-your-work](https://github.com/showyourwork/showyourwork) does.  Do you know how journals are handling [show-your-work](https://github.com/showyourwork/showyourwork) submitted manuscripts?

I too have been invested in `LaTeX` for many years, so I'm probably going to keep with it for articles, but my main purpose for exploring [Quarto](https://quarto.org/) is to extend my note taking capabilities and presentations. I usually use beamer to make presentations but I never quite liked the experience. I'm experimenting with using quarto instead of jupyter notebooks and then, when I need it, creating [revealjs](https://quarto.org/docs/presentations/revealjs/) presentations. What I'm hopeful about is that Quarto/Pandoc will also make it easy to convert `revealjs` to `pptx` and `pdf`; although I'm skeptical there will be much success converting to `pptx`.
